pkg/machines/index.es6

```diff
@@ -51,3 +46,33 @@ export function appMain() {
     // initiate data retrieval
     store.dispatch(initDataRetrieval());
 }
+
+/**
+ * Start the application
+ */
+export function appMain() {
+    logDebug('index.es6: initial state: ' + JSON.stringify(store.getState()));
+
+    let client = cockpit.dbus("org.freedesktop.DBus");
+
+    client.call("/org/freedesktop/DBus", "org.freedesktop.DBus",
+                "ListActivatableNames")
+            .done(services => {
+                if (services[0].includes("org.libvirt")) {
+                    logDebug("index.es6: Setting libvirt-dbus as virt provider.");
+                    setVirtProvider(LibvirtDbus);
+                } else {
+                    logDebug("index.es6: Setting virsh as virt provider.");
+                    setVirtProvider(Libvirt);
+                }
+                client.close();
+                renderApp();
+            })
+            .fail(exception => {
+                console.warn("Could not get a list of services from DBus.", exception);
+                logDebug("index.es6: Setting virsh as virt provider.");
+                setVirtProvider(Libvirt);
+                client.close();
+                renderApp();
+            });
+}
```

Comment:
I would change this code in a way that I would introduce a helper function called `checkLibvirtDBus()` or something like that and use that function in `appMain()` to decide what provider should be used.  That way there would be no need to create `renderApp()` function and have the call on multiple places.